experimental/packages/opentelemetry-sdk-web/src/sdk.ts

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ContextManager,
+  TextMapPropagator,
+} from '@opentelemetry/api';
+import {
+  InstrumentationOption,
+  registerInstrumentations,
+} from '@opentelemetry/instrumentation';
+import {
+  Detector,
+  DetectorSync,
+  detectResourcesSync,
+  IResource,
+  browserDetector,
+  Resource,
+  ResourceDetectionConfig,
+} from '@opentelemetry/resources';
+import {
+  BatchSpanProcessor,
+  SpanProcessor,
+} from '@opentelemetry/sdk-trace-base';
+import {
+  WebTracerConfig,
+  WebTracerProvider,
+} from '@opentelemetry/sdk-trace-web';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import { WebSDKConfiguration } from './types';
+import { SessionIdSpanProcessor } from './SessionIdSpanProcessor';
+
+/** This class represents everything needed to register a fully configured OpenTelemetry Web SDK */
+
+export class WebSDK {
```

Comment:
What do you think of pulling the common pieces of this + the `NodeSDK` into an abstract class that takes a few type arguments? It could ensure that these 2 don't drift over time

Reply:
There is definitely some overlap, e.g. setting service name, running resource detectors and registering instrumentations. But there are also many differences: Node SDK has metrics and logs while Web does not, Web will have events while Node will not (at least for now), NodeTracerProvider vs WebTracerProvider, handling env variables in the Node SDK.

It seems to me that there are a lot of differences, and it might be tricky to come up with an abstract class that works for both (I might be wrong though). I would prefer to get this merged and optimize later.

Reply:
Fair enough, fine as is!